elastic/indexer/elastic-indexer.go

```diff
@@ -52,20 +55,84 @@ func New(esHosts []string, esIndex, esMapping, kafkaAddress, changesTopic, kafka
 }
 
 func (i *Indexer) runStep() error {
-	m, err := i.kReader.FetchMessage(context.Background())
+	messages, err := i.readMessageBlock(5*time.Second, i.bulkSize)
+	log.Println("Messages Bulk: ", len(messages))
+	if err != nil {
+		return err
+	}
+
+	if len(messages) == 0 {
+		return nil
+	}
+
+	var bulkBody string
+	bulkDocumentIdKafkaMessages := make(map[string]kafka.Message)
+	for _, message := range messages {
+
+		changeMessage := &changestream.ChangeMessage{}
+		if err := json.Unmarshal(message.Value, changeMessage); err != nil {
+			return err
+		}
+		bulkOperation, err := i.indexFunc(changeMessage)
+
+		if err != nil {
+			return err
+		}
+
+		bulkDocumentIdKafkaMessages[bulkOperation.DocumentId] = message
+		bulkBody += bulkOperation.BulkOperation
+	}
+
+	bulkResponse, err := i.esClient.Bulk(strings.NewReader(bulkBody), i.esClient.Bulk.WithIndex(i.esIndex))
 	if err != nil {
 		return err
 	}
+	log.Println("Result Messages Bulk: ", bulkResponse.Status())
+
+	body, err := ioutil.ReadAll(bulkResponse.Body)
 
-	changeMessage := &changestream.ChangeMessage{}
-	if err := json.Unmarshal(m.Value, changeMessage); err != nil {
+	if err != nil {
 		return err
 	}
 
-	if err := i.indexFunc(i.esClient, changeMessage); err != nil {
+	var result bulkResult
+	err = json.Unmarshal(body, &result)
+
+	if err != nil {
 		return err
 	}
-	return i.kReader.CommitMessages(context.Background(), m)
+
+	log.Println("BulkResultItem: ", len(result.Items))
+	log.Println("BulkResult: ", string(body))
+	for _, bulkResultOperation := range result.Items {
+
+		if bulkResultOperation.Index.ID != "" {
+			err := i.commitKafkaMessages(bulkResultOperation.Index.Status, bulkResultOperation.Index.ID, bulkDocumentIdKafkaMessages)
```

Comment:
As said in person: 
Be careful with committing when one of the items contains an error